projects/packages/stats/tests/php/test-options.php

```diff
@@ -0,0 +1,388 @@
+	 *
+	 * @dataProvider statsGetOptionDataProvider
+	 *
+	 * @param  string $option_name  The Stats option name.
+	 * @param  mixed  $option_value The expected Stats option value.
+	 */
+	public function test_get_option( $option_name, $option_value ) {
+		$this->assertSame( $option_value, Options::get_option( $option_name ) );
+	}
+
+	/**
+	 * Data provider for test_set_options.
+	 *
+	 * @return array
+	 */
+	public function statsSetOptionsDataProvider() {
+		return array(
+			'not array'            => array(
+				'set_options'    => null,
+				'result'         => false,
+				'stored_options' => false,
+			),
+			'set only count roles' => array(
+				'options'        => array(
+					'count_roles' => array(),
+				),
+				'result'         => true,
+				'stored_options' => array(
+					'admin_bar'    => true,
+					'roles'        => array(
+						'administrator',
+					),
+					'count_roles'  => array(),
+					'do_not_track' => true,
+					'blog_id'      => 1234,
+					'version'      => '9',
+				),
+			),
+			'set version'          => array(
+				'options'        => array(
+					'version' => 'dummy',
+				),
+				'result'         => true,
+				'stored_options' => array(
+					'admin_bar'    => true,
+					'roles'        => array(
+						'administrator',
+					),
+					'count_roles'  => array(),
+					'do_not_track' => true,
+					'blog_id'      => 1234,
+					'version'      => '9',
+				),
+			),
+			'set blog blog_id'     => array(
+				'options'        => array(
+					'blog_id' => 999,
+				),
+				'result'         => true,
+				'stored_options' => array(
+					'admin_bar'    => true,
+					'roles'        => array(
+						'administrator',
+					),
+					'count_roles'  => array(),
+					'do_not_track' => true,
+					'blog_id'      => 1234,
+					'version'      => '9',
+				),
+			),
+			'multiple options'     => array(
+				'options'        => array(
+					'admin_bar'    => false,
+					'roles'        => array(
+						'administrator',
+						'editor',
+					),
+					'count_roles'  => array(
+						'administrator',
+					),
+					'do_not_track' => false,
+				),
+				'result'         => true,
+				'stored_options' => array(
+					'admin_bar'    => false,
+					'roles'        => array(
+						'administrator',
+						'editor',
+					),
+					'count_roles'  => array(
+						'administrator',
+					),
+					'do_not_track' => false,
+					'blog_id'      => 1234,
+					'version'      => '9',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Test for Options::set_options
+	 *
+	 * @dataProvider statsSetOptionsDataProvider
+	 *
+	 * @param mixed $set_options     The Stats options to set.
+	 * @param bool  $result          The expected result.
+	 * @param mixed $stored_options  The expected Stats stored options.
+	 */
+	public function test_set_options( $set_options, $result, $stored_options ) {
+		$this->assertSame( $result, Options::set_options( $set_options ) );
+		$this->assertSame( $stored_options, get_option( 'stats_options' ) );
```

Comment:
This second assert can confuse, because it actually checks `get_option`. If `get_option` is broken then this test that should check `set_options` will incorrectly fail. Is there another way to check the `$stored_options`.

Reply:
`get_option` is a WP core function though.. And we are using WordDBless so I couldn't make a DB query 🤔

Reply:
Oh, it makes sense then. I totally missed that `get_option` is the core function 🤦🏼.